Analyticsv3Sample/AnalyticsSampleProject/StandardGoogleAuthSample.cs

```diff
@@ -0,0 +1,239 @@
+                                             AnalyticsService.Scope.AnalyticsEdit,                   // Edit Google Analytics management entities
+                                             AnalyticsService.Scope.AnalyticsManageUsers,            // Manage Google Analytics Account users by email address
+                                             AnalyticsService.Scope.AnalyticsManageUsersReadonly,    // View Google Analytics user permissions
+                                             AnalyticsService.Scope.AnalyticsProvision,              // Create a new Google Analytics account along with its default property and view
+                                             AnalyticsService.Scope.AnalyticsReadonly};              // View your Google Analytics data
+            try
+            {
+                UserCredential credential;
+                using (var stream = new FileStream(clientSecretJson, FileMode.Open, FileAccess.Read))
+                {
+                    string credPath = System.Environment.GetFolderPath(System.Environment.SpecialFolder.Personal);
+                    credPath = Path.Combine(credPath, ".credentials/Analytics");
+
+                    credential = GoogleWebAuthorizationBroker.AuthorizeAsync(
+                        GoogleClientSecrets.Load(stream).Secrets,
+                        scopes,
+                        userName,
+                        CancellationToken.None,
+                        new FileDataStore(credPath, true)).Result;
+                    Console.WriteLine("Credential file saved to: " + credPath);
+                }
+
+                // Create Analytics API service.
+                var service = new AnalyticsService(new BaseClientService.Initializer()
+                    {
+                        HttpClientInitializer = credential,
+                        ApplicationName = "Analytics Authentication Sample",
+                    });
+                return service;
+            }
+            catch (Exception ex)
+            {
+
+                Console.WriteLine(ex.InnerException);
+                throw ex;
+
+            }
+        }
+        /// <summary>
+        /// Authenticate to Google Using Oauth2
+        /// Documentation https://developers.google.com/accounts/docs/OAuth2
+        /// </summary>
+        /// <param name="clientId">From Google Developer console https://console.developers.google.com</param>
+        /// <param name="clientSecret">From Google Developer console https://console.developers.google.com</param>
+        /// <param name="userName">The user to authorize.</param>
+        /// <returns>a valid AnalyticsService</returns>
+        public static AnalyticsService AuthenticateOauth(string clientId, string clientSecret, string userName)
+        {
+            if (string.IsNullOrEmpty(clientId))
+                throw new Exception("clientId is required.");
```

Comment:
Consider changing your exception to Argument Exception (https://msdn.microsoft.com/en-us/library/system.argumentexception(loband).aspx)